### PR TITLE
Index Page to ListView Routing With the Search Api

### PR DESCRIPTION
**Pull request by MayankTamakuwala**
Integrated Search API in the frontend to the Index Page and made routing to the ListView.js.

### CODE REVIEW

File: api/autocomplete.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const autocompleteSearch = async (itemInput) => {
```
Comment (AlfredoSequeida) [inline api/autocomplete.js:21]: let's add some comments for this function with @param and @return docs

**Review by AlfredoSequeida — DISMISSED** [review]
Thank you and @dani5612 for working on this. The only request I have is adding a missing function comment. After that addition, we should be good to merge!

**Review by AlfredoSequeida — APPROVED** [review]
Looks good!

**Review by SeaWalks — APPROVED** [review]
Looks good! Cool to see old faker stuff getting replaced by real data.

**Pull request merged**